word_guess.rb

```diff
@@ -0,0 +1,269 @@
+    @guess = "" # initializes w/ empty guess
+    @wrong_count = 5 # Maybe will change this difficulty level
+    @difficulty = "" # init with empty difficulty so player can choose
+    @pick_word = {easy: ["dog", "cat", "star"].sample,
+      medium: ["pizza", "input", "coder"].sample,
+      hard: ["birthday", "computation", "ubiquitous"].sample} # selects a word for each difficulty
+    @word = "" # placeholder for word we will be guessing
+    @alphabet_array = ("A".."Z").to_a # letters to check guesses against
+    @used_letters = [] # place holder for guessed letters
+    @word_so_far = "" # placeholder to hold the word as it is guessed
+    @bad_prize = ["stinky socks", "moldy tofu", "rotten eggs", "more homework", "dirty diapers"].sample
+    @good_prize = ["elephant hotdog", "invisible pineapple", "expensive whiskey", "customized puppet of yourself"].sample
+  end
+
+
+  # this method enables playing the game
+  def play_the_game
+    pick_a_word
+    update_board
+    until @wrong_count == 0
+      guess
+    end
+    lose_game
+  end
+
+  private
+
+  # this method defines the difficulty level and sets the word to guess
+  def pick_a_word
+    if @word == ""
+      puts "What difficulty would you like to play at? There's easy(E), medium(M), or hard(H)!"
+      print "Difficulty: "
+      @difficulty = gets.chomp.upcase
+      if @difficulty == "E" || @difficulty == "EASY"
+        @word = @pick_word[:easy]
+        @word = @word.upcase
+      elsif @difficulty == "M" || @difficulty == "MEDIUM"
+        @word = @pick_word[:medium]
+        @word = @word.upcase
+      elsif @difficulty == "H" || @difficulty == "HARD"
+        @word = @pick_word[:hard]
+        @word = @word.upcase
+      else
+        # if they try to enter anything except a difficulty level
+        # the game throws up and goes home
+        puts "I don't know what you want, please start over!"
+        exit
+      end
+      @word_so_far = "_" * @word.length
+    end
+  end
+
+  # this method operates getting guesses and checking them.
+  def guess
+    get_input
+    check_letter
+    if word_done?
+      win_game
+    end
+    if !more_tries?
+      lose_game
+    end
+  end
+
+  # this method actually gets the input
+  def get_input
+    puts "What letter do you guess?"
+    @guess = gets.chomp.upcase
+
+    # this checks if you guessed the right word.
+    while @guess.length > 1
+      if @guess == @word.upcase
+        win_game
+      else
+        #second time we use this- refactor into a method?
+        @wrong_count -= 1
+        puts "You have made a terrible guess!"
+        puts "How could you pick that letter?!"
+        @used_letters.push(@guess)
+        update_board
+        puts "You have #{ @wrong_count } guesses left."
+        @guess = gets.chomp.upcase
+      end
+    end
+
+    # this checks if the guess is a single character and if that character is a letter!
+    while @guess.length != 1 || @alphabet_array.include?(@guess) == false
```

Comment:
I wonder if this logic should happen before the logic on line 78. I wouldn't want to check this "bad" guess if it wasn't an alphabetical character.